### PR TITLE
app-shells/autojump: don't install into /usr/local

### PR DESCRIPTION
**Pull request by lukipuki**
As pointed out by @lkraav in #66, packages shouldn't install into `/usr/local`.

### CODE REVIEW

**Comment by mgorny** [comment]
You should definitely remove the previous version (as in `git rm`).

**Comment by lukipuki** [comment]
@mgorny done

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+PYTHON_COMPAT=( python{2_7,3_3,3_4} )
+
+inherit python-r1 python-utils-r1 vcs-snapshot
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Do not inherit python-utils-r1 explicitly.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+DESCRIPTION="change directory command that learns"
+HOMEPAGE="https://github.com/joelthelion/autojump"
+SRC_URI="https://github.com/joelthelion/${PN}/archive/release-v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~ppc ~ppc64"
+IUSE="bash-completion python test zsh-completion"
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: USE=*-completion is deprecated and should not be used anymore. Completions are to be installed unconditionally.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	zsh-completion? ( app-shells/zsh app-shells/gentoo-zsh-completions )"
+DEPEND="test? ( dev-python/flake8 dev-python/tox )"
+
+src_prepare() {
+	sed -e "s: \(/etc/profile.d\): \"${EPREFIX}\1\":" \
+		-i bin/autojump.sh || die
+}
+
+src_compile() {
+	true
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: `:` is more common.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+src_install() {
+	dobin bin/autojump
+
+	insinto /etc/profile.d
+	doins bin/${PN}.sh
+
+	insinto /usr/share/"${PN}"/
+	if use bash-completion ; then
+		doins bin/${PN}.bash
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: How does that work, exactly? That's not the common location for completions.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	if use zsh-completion ; then
+		doins bin/${PN}.zsh
+		insinto /usr/share/zsh/site-functions
+		doins bin/_j
+	fi
+
+	if use python ; then
+		python_foreach_impl python_domodule tools/autojump_ipython.py bin/autojump_argparse.py bin/autojump_data.py bin/autojump_utils.py
+
+		einfo "This tool provides \"j\" for ipython, please add"
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: If you used outer `'`, you wouldn't have to resort to escapes.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 1 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Fixed

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 2 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Fixed

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 3 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Fixed

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 5 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Fixed

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 4 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: No idea, I just copied the old ebuild and changed two lines. Maybe I'll investigate later if I have time.

**Comment by lkraav** [comment]
https://github.com/gentoo/gentoo/blob/master/dev-vcs/tig/tig-2.1.1.ebuild shows an up-to-date `bash-completion` implementation

**Comment by lukipuki** [comment]
@lkraav fixed, although I use zsh and can't test it

File: app-shells/autojump/autojump-22.2.4-r2.ebuild (hunk 4 above)
Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: I've changed it to `newbashcomp` in 3c5fbd4. I'm not a bash user, so I can't test it unfortunately.

File: app-shells/autojump/autojump-22.2.4-r2.ebuild [hunk 6]
```diff
@@ -42,21 +38,17 @@ src_install() {
...
-		doins bin/${PN}.bash
-	fi
-
-	if use zsh-completion ; then
-		doins bin/${PN}.zsh
-		insinto /usr/share/zsh/site-functions
-		doins bin/_j
-	fi
+	newbashcomp bin/${PN}.bash ${PN}
+	insinto /usr/share/"${PN}"/
```
Comment (mgorny) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Now I'm being picky but you're quoting PN inconsistently.

Comment (hasufell) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: PN is not allowed to have whitespaces anyway (unless you overwrite it with funny stuff), so just remove all quotes

Comment (lukipuki) [inline app-shells/autojump/autojump-22.2.4-r2.ebuild]: Fixed

**Comment by monsieurp** [comment]
merged :+1:

**Pull request closed without merging**